ampd/src/handlers/solana_verify_verifier_set.rs

```diff
@@ -0,0 +1,377 @@
+#[try_from("wasm-verifier_set_poll_started")]
+struct PollStartedEvent {
+    verifier_set: VerifierSetConfirmation,
+    poll_id: PollId,
+    source_chain: ChainName,
+    expires_at: u64,
+    participants: Vec<TMAddress>,
+}
+
+pub struct Handler<C: SolanaRpcClientProxy> {
+    verifier: TMAddress,
+    voting_verifier_contract: TMAddress,
+    rpc_client: C,
+    solana_gateway_domain_separator: [u8; 32],
+    latest_block_height: Receiver<u64>,
+    event_match_context: MatchContext,
+}
+
+impl<C: SolanaRpcClientProxy> Handler<C> {
+    pub async fn new(
+        verifier: TMAddress,
+        voting_verifier_contract: TMAddress,
+        rpc_client: C,
+        latest_block_height: Receiver<u64>,
+    ) -> Self {
+        let domain_separator = rpc_client
+            .get_domain_separator()
+            .await
+            .expect("cannot start handler without fetching domain separator for Solana");
+        let event_match_context = MatchContext::new(&axelar_solana_gateway::ID.to_string());
+
+        Self {
+            verifier,
+            solana_gateway_domain_separator: domain_separator,
+            event_match_context,
+            voting_verifier_contract,
+            rpc_client,
+            latest_block_height,
+        }
+    }
+
+    fn vote_msg(&self, poll_id: PollId, vote: Vote) -> MsgExecuteContract {
+        MsgExecuteContract {
+            sender: self.verifier.as_ref().clone(),
+            contract: self.voting_verifier_contract.as_ref().clone(),
+            msg: serde_json::to_vec(&ExecuteMsg::Vote {
+                poll_id,
+                votes: vec![vote],
+            })
+            .expect("vote msg should serialize"),
+            funds: vec![],
+        }
+    }
+
+    async fn fetch_message(
+        &self,
+        msg: &VerifierSetConfirmation,
+    ) -> Option<(solana_sdk::signature::Signature, UiTransactionStatusMeta)> {
+        let signature = solana_sdk::signature::Signature::from(msg.message_id.raw_signature);
+        self.rpc_client
+            .get_tx(&signature)
+            .await
+            .map(|tx| (signature, tx))
+    }
+}
+
+#[async_trait]
+impl<C: SolanaRpcClientProxy> EventHandler for Handler<C> {
+    type Err = Error;
+
+    async fn handle(&self, event: &events::Event) -> Result<Vec<Any>> {
+        if !event.is_from_contract(self.voting_verifier_contract.as_ref()) {
+            return Ok(vec![]);
+        }
+
+        let PollStartedEvent {
+            poll_id,
+            source_chain,
+            expires_at,
+            participants,
+            verifier_set,
+        } = match event.try_into() as error_stack::Result<_, _> {
+            Err(report) if matches!(report.current_context(), EventTypeMismatch(_)) => {
+                return Ok(vec![])
+            }
+            event => event.change_context(Error::DeserializeEvent)?,
+        };
+
+        if !participants.contains(&self.verifier) {
+            return Ok(vec![]);
+        }
+
```

Comment:
should check the source chain

Reply:
Let's follow conversation here: https://github.com/axelarnetwork/axelar-amplifier/pull/744#discussion_r1936064825

Reply:
https://github.com/axelarnetwork/axelar-amplifier/pull/744#discussion_r2001569525